vscode-lean4/src/infoview.ts

```diff
@@ -389,10 +389,15 @@ export class InfoProvider implements Disposable {
         } else {
             editor = window.activeTextEditor;
             if (!editor) { // sometimes activeTextEditor is null.
-                editor = window.visibleTextEditors[0];
+                editor = window.visibleTextEditors.find(e => e.document.languageId === 'lean4');
             }
         }
-        if (!editor) return;
+        if (!editor) {
+            // user must have switch away from any lean source file in which case we don't know
+            // what to do here.  TODO: show a popup error?  Or should we use the last uri used in
+            // sendPosition and automatically activate that editor?
```

Comment:
The other question is what should happen if the user has multiple visible editors with Lean files (like I often do).